index.js

```diff
@@ -70,21 +71,27 @@ function lint(textEditor) {
 			};
 		}
 
+		// taken from linter-eslint
+		let range;
+		const msgLine = x.line - 1;
+		if (typeof x.endColumn === 'number' && typeof x.endLine === 'number') {
+			const msgCol = Math.max(0, x.column - 1);
+			range = [[msgLine, msgCol], [x.endLine - 1, x.endColumn - 1]];
+		} else if (typeof x.line === 'number' && typeof x.column === 'number') {
+			// We want msgCol to remain undefined if it was initially so
+			// `rangeFromLineNumber` will give us a range over the entire line
+			const msgCol = typeof x.column === 'undefined' ? x.column : x.column - 1;
+			range = rangeFromLineNumber(textEditor, msgLine, msgCol);
```

Comment:
`linter-eslint` [catches](https://github.com/AtomLinter/linter-eslint/blob/3b9e2dbc34f8b30fbafdcbca98382dd10c27530e/src/main.js#L216-L221) the error and makes it more presentable (showing rule name, originally requested position, and message), Atom will simply show what `rangeFromLineNumber` throws, which is admittedly somewhat cryptic to users:

`Column start (x) greater than line length (y) for line z`

As a note, see [here](https://github.com/AtomLinter/linter-eslint/issues/735#issuecomment-254647063) for my current plans on making this less of a pain for users, I just haven't had time to implement it yet.